buildSrc/src/main/java/edu/wpi/gripgenerator/settings/DefinedParamType.java

```diff
@@ -41,7 +41,7 @@ public boolean isOutput() {
     }
 
     private final String type;
-    private DefinedParamState state;
+    private Optional<DefinedParamState> state;
```

Comment:
It's less `Optional` and more undefined until requested. This is better than using null.